python/cudf_polars/cudf_polars/callback.py

```diff
@@ -226,6 +227,15 @@ def validate_config_options(config: dict) -> None:
         config.get("parquet_options", {})
     )
 
+    # Validate executor_options
+    executor = config.get("executor", "pylibcudf")
+    if executor == "dask-experimental":
+        unsupported = config.get("executor_options", {}).keys() - {"num_rows_threshold"}
+    else:
+        unsupported = config.get("executor_options", {}).keys()
+    if unsupported:
+        raise ValueError(f"Unsupported executor_options for {executor}: {unsupported}")
```

Comment:
@pentschev - I'm open to adding a new `Executor` class system (in a different PR). For now, this PR just assumes that the "executor_options" arguments are specific to whatever the "executor" argument is. (which is effectively the same as "nesting"). If we add an `Executor` class, this validation logic could just move into that class.

Reply:
No objections from me, my proposal was more long-term targeted. 🙂